### PR TITLE
Physical_oM: foundation objects added

### LINKED ISSUE #566: Physical_oM: implement PileCap class
#### Description:
Could be helpful, e.g. on Revit interop.

### LINKED ISSUE #1155: Physical_oM: Creation of more foundations objects
#### Description:

Creation of Footing, PileCap and GradeBeam or similarly named Foundations elements. 

These types of objects make up a very large percentage of LCA-related impacts and would be very helpful if integrated in our typical workflows as a simple use case. 

Has any previous thought gone into the creation of these objects for integration in other workflows? 
@IsakNaslundBh @pawelbaran 

FYI: @mchaf

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #566
Closes #1155

### Test files

Just object creation.

### CODE REVIEW

**Comment by pawelbaran** [comment]
Apologies @FraserGreenroyd for raising this PR during the dead week, but I wanted to do it fresh after the workshop held yesterday. Also put `WIP` label for now - let's not merge before we gather enough feedback (@peterjamesnugent structures, @michaelhoehn LCA, @al-fisher general if interested).

**Comment by FraserGreenroyd** [comment]
> Apologies @FraserGreenroyd for raising this PR during the dead week, but I wanted to do it fresh after the workshop held yesterday. Also put `WIP` label for now - let's not merge before we gather enough feedback (@peterjamesnugent structures, @michaelhoehn LCA, @al-fisher general if interested).

No worries @pawelbaran , I've done all the automation needed from my side so planning week can continue but no issues with new PRs at this point - just want us to give some focus to existing PRs to merge them out but otherwise no need to apologise for this one 😄

**Comment by FraserGreenroyd** [comment]
During mid-sprint touchdown 23.10.23, requesting feedback from @michaelhoehn and @al-fisher to keep momentum on this during the next couple of weeks.

**Review by peterjamesnugent — CHANGES_REQUESTED** [review]
Where object types are referenced, I have capitalised. One small change to the description of `Pile`.

I will raise the Engine PR shortly and these can all be merged together.

**Comment by pawelbaran** [comment]
Thanks @peterjamesnugent, I applied your suggestions 👍

**Review by peterjamesnugent — DISMISSED** [review]
Reviewed by inspection and previous comments implemented.

**Comment by peterjamesnugent** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright-compliance
@BHoMBot check ready-to-merge

File: Physical_oM/Elements/Pile.cs [hunk 1]
```diff
@@ -20,25 +20,24 @@
...
-
 using BH.oM.Base;
 using BH.oM.Geometry;
 using BH.oM.Physical.FramingProperties;
+using System.ComponentModel;
 
 namespace BH.oM.Physical.Elements
 {
+    [Description("Physical representation of a pile.")]
     public class Pile : BHoMObject, IFramingElement
```
Comment (peterjamesnugent) [inline Physical_oM/Elements/Pile.cs:31]: Should this inherit `IElementM` for mass methods, and `IElement1D`?

Comment (pawelbaran) [reply Physical_oM/Elements/Pile.cs:31]: `IFramingElement` implements both interfaces you mentioned @peterjamesnugent 👍

**Review by peterjamesnugent — DISMISSED** [review]
Approved based on recent comments, as disucssed with @pawelbaran to make `Pile`s in `PileFoundation` read only.

**Comment by peterjamesnugent** [comment]
@BHoMBot check required
@BHoMBot check copyright-compliance
@BHoMBot check ready-to-merge

**Comment by pawelbaran** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright-compliance

**Review by peterjamesnugent — APPROVED** [review]
Based on previous review and CI/CD check being actioned.

**Comment by peterjamesnugent** [comment]
@BHoMBot check ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Comment by pawelbaran** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**